src/quality_result_gui/ui/quality_errors_dock.py

```diff
@@ -317,6 +318,7 @@ def _on_current_item_changed(
                 [error_feature], preserve_scale=False
             )
 
+        self.quality_error_selected.emit(error_feature, self.source_button_event)
```

Comment:
Would it be better to emit QualityError instance as it contains also feature id etc. which could be used for other things? ErrorFeature class is meant mainly for visualizing data (it should probably be renamed...) so it doesn't contain that much information (for this reason feature_type is missing).

How about renaming `_get_error_feature_from_row` to `_get_quality_error_from_row` and make it return `Optional[QualityError]` instead of  `Optional[ErrorFeature]`? Then create error_feature for visualizer before click event is checked and emit quality error in the end. This way no changes to visualizer are necessary.

```python
@log_if_fails
 def _on_current_item_changed(
    self, current_index: QModelIndex, previous_index: QModelIndex
) -> None:
    quality_error = self._get_quality_error_from_row(current_index)

    if quality_error is None:
        return
    
    # Create error feature for visualizer
    error_feature = ErrorFeature(
            quality_error.unique_identifier,
            quality_error.priority,
            quality_error.geometry,
            self._crs,
        )
    # Right button behaviour
    ...

    self.quality_error_selected.emit(quality_error, self.source_button_event)
```

Reply:
Yes it is better to send QualityError instead of ErrorFeature in mouse event, did the proposed code changes